msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -163,39 +151,122 @@ public AcquireTokenResult acquireToken(AcquireTokenOperationParameters parameter
     }
 
     /**
-     * Get the intent for the broker interactive request
-     *
-     * @param parameters
-     * @return
+     * Info of a broker operation to be performed with available strategies.
+     */
+    private interface BrokerOperationInfo<T extends OperationParameters, U> {
+        /**
+         * Performs this broker operation in this method with the given IMicrosoftAuthService.
+         */
+        @Nullable
+        U perform(BrokerBaseStrategy strategy, T parameters) throws Exception;
+
+        /**
+         * Name of the task (for logging purposes).
+         */
+        String getMethodName();
+
+        /**
+         * Name of the telemetry API event associated to this strategy task.
+         * If this value returns null, no telemetry event will be emitted.
+         */
+        @Nullable
+        String getTelemetryApiName();
+
+        /**
+         * A method that will be invoked before the success event is emitted.
+         * If the calling operation wants to put any value in the success event, put it here.
+         */
+        void putValueInSuccessEvent(ApiEndEvent event, U result);
+    }
+
+    /**
+     * A generic method that would initialize and iterate through available strategies.
+     * It will return a result immediately if any of the strategy succeeds, or throw an exception if all of the strategies fails.
      */
-    private Intent getBrokerAuthorizationIntent(@NonNull final AcquireTokenOperationParameters parameters) throws ClientException {
-        final String methodName = ":getBrokerAuthorizationIntent";
-        helloBroker(parameters);
+    private <T extends OperationParameters, U> U performStrategies(@NonNull final T parameters,
```

Comment:
`performStrategies` doesn't seem to fit here . I think we need to rename this , how about `invokeBrokerOperation` or `invokeBrokerApi` ?